gef-ui/server/controllers/unissued-facilities/index.js

```diff
@@ -0,0 +1,516 @@
+      });
+    }
+  } else if (coverEndDateIsPartiallyComplete) {
+    let msg = 'Cover end date must include a ';
+    const dateFieldsInError = [];
+    if (!coverEndDateDay) {
+      msg += 'day ';
+      dateFieldsInError.push('coverEndDate-day');
+    }
+    if (!coverEndDateMonth) {
+      msg += !coverEndDateDay ? ' and month ' : 'month ';
+      dateFieldsInError.push('coverEndDate-month');
+    }
+    if (!coverEndDateYear) {
+      msg += !coverEndDateDay || !coverEndDateMonth ? 'and year' : 'year';
+      dateFieldsInError.push('coverEndDate-year');
+    }
+
+    aboutFacilityErrors.push({
+      errRef: 'coverEndDate',
+      errMsg: msg,
+      subFieldErrorRefs: dateFieldsInError,
+    });
+  }
+
+  if (body.facilityName && body.facilityName.length > 30) {
+    aboutFacilityErrors.push({
+      errRef: 'facilityName',
+      errMsg: 'Facility name cannot be more than 30 characters in length',
+    });
+  }
+
+  if (/[^A-Za-z0-9 .,:;'-]/.test(body.facilityName)) {
+    aboutFacilityErrors.push({
+      errRef: 'facilityName',
+      errMsg: 'Facility name must only include letters a to z, full stops, commas, colons, semi-colons, hyphens, spaces and apostrophes',
+    });
+  }
+
+  if (issueDateIsFullyComplete) {
+    issueDate = set(new Date(), { year: issueDateYear, month: issueDateMonth - 1, date: issueDateDay });
+  }
+
+  if (coverStartDateIsFullyComplete) {
+    coverStartDate = set(new Date(), { year: coverStartDateYear, month: coverStartDateMonth - 1, date: coverStartDateDay });
+  }
+
+  if (coverEndDateIsFullyComplete) {
+    coverEndDate = set(new Date(), { year: coverEndDateYear, month: coverEndDateMonth - 1, date: coverEndDateDay });
+  }
+
+  if (coverStartDateIsFullyComplete && coverEndDateIsFullyComplete) {
+    if (coverEndDate < coverStartDate) {
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'Cover end date cannot be before cover start date',
+      });
+    }
+  }
+
+  if (issueDateIsFullyComplete && coverEndDateIsFullyComplete) {
+    if (coverEndDate < issueDate) {
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'Cover end date cannot be before the issue date',
+      });
+    }
+  }
+
+  if (issueDateIsFullyComplete && coverStartDateIsFullyComplete) {
+    if (coverStartDate < issueDate) {
+      aboutFacilityErrors.push({
+        errRef: 'coverStartDate',
+        errMsg: 'Cover start date cannot be before the issue date',
+      });
+    }
+  }
+
+  // Regex tests to see if value is a number only
+  const digitsRegex = /^[0-9]*$/;
+  if (body.monthsOfCover) {
+    if (!digitsRegex.test(body.monthsOfCover)) {
+      aboutFacilityErrors.push({
+        errRef: 'monthsOfCover',
+        errMsg: 'You can only enter numbers',
+      });
+    }
+    if (body.monthsOfCover > 999) {
+      aboutFacilityErrors.push({
+        errRef: 'monthsOfCover',
+        errMsg: 'You can only enter a maximum of 999 months cover',
+      });
+    }
+  }
+
```

Comment:
Lots and lots of lines for validation, can we move to it's own validation file please? Can have it returning the errors array

Reply:
validation.js within unissued-facilities? or a validation controller?

Reply:
done